fineract-provider/src/main/java/org/apache/fineract/infrastructure/codes/service/CountryNamesTranslationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.codes.service;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Properties;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CountryNamesTranslationServiceImpl {
+
+    public String retrieveTranslationByCode(String value) {
+        Properties countryProperties = new Properties();
```

Comment:
With this implementation, the translation is read every time we try to translate a country code. Let's load the translations on startup time and work with that.

On the other hand, please rename the method since it doesn't indicate anything specific to countries but as a general translation method.